corehq/apps/user_importer/helpers.py

```diff
@@ -252,6 +252,25 @@ def update_locations(self, location_codes, domain_info):
             else:
                 self.logger.add_info(UserChangeMessage.primary_location_removed())
 
+    def update_user_groups(self, domain_info, group_names):
+        try:
+            old_group_ids = set()
+            for group in domain_info.group_memoizer.by_user_id(self.user.user_id):
+                old_group_ids.add(group.get_id)
+                if group.name not in group_names:
+                    group.remove_user(self.user)
+
+            new_groups = {}
+            for group_name in group_names:
+                group = domain_info.group_memoizer.by_name(group_name)
+                group.add_user(self.user, save=False)
+                new_groups[group.get_id] = group
+
+            if set(new_groups) != old_group_ids:
+                self.logger.add_info(UserChangeMessage.groups_info(list(new_groups.values())))
+        except Exception as e:
+            return str(e)
```

Comment:
what exceptions can get raised here? I had imagined that the error would come from the `BulkSaveError` when calling `domain_info.group_memoizer.save_all()`.

Thinking about this a bit more I can see why that was in the finally block - so that we don't re-save the groups for after each user row is processed.